tests/e2etest/GenericTest/src/main/java/org/astarteplatform/devicesdk/tests/e2etest/utilities/GenericDeviceMockData.java

```diff
@@ -0,0 +1,10 @@
+package org.astarteplatform.devicesdk.tests.e2etest.utilities;
```

Comment:
Do you really need this interface and `GenericDeviceMockDataFactory` class? I think you can use `GenericDeviceSingleton` directly,  for e2etest, it is a simple use case for testing.

Reply:
My factory pattern is for call mock data and a singleton class, which I think is an easy way to read the code.